src/simulations/data_manager.py

```diff
@@ -203,7 +203,7 @@ def _active_find_feasible_data(
 
         if all(data_feasibles) and opt_needed and (new_data_needed or self.force_optimisation):
             try:
-                res = self.optimiser.solve(self.prm)
+                res = self.optimiser.solve(self.prm, epoch)
```

Comment:
add timer before and after` res = self.optimiser.solve(self.prm)` here directly. 

I would not do the recording in the `optimiser`. 

Maybe create a `self.times_optimisation` attribute for the `data_manager` object when initialising. Then append with the duration after each `res = self.optimiser.solve(self.prm)`. Then at the very end of the experiment when you do `record.save(end_of='end')` you can save the mean and std of all the values in that list `self.explorer.data. self.times_optimisation`.

Reply:
In the function `run`, where we `record.save`, I don't perfectly understand how I would access the class `DataManager` where I have saved my list of `times_optimisation`.

Reply:
`self.explorer.data.times_optimisation`

Reply:
Thanks a lot!